sycl/test-e2e/format.py

```diff
@@ -87,19 +76,60 @@ def parseTestScript(self, test):
 
         return script
 
-    def getMatchedFromList(self, features, alist):
+    def getMissingRequiredFeaturesFromList(
+        self, features, requires, build_only_mode=False
+    ):
         try:
             return [
-                item for item in alist if BooleanExpression.evaluate(item, features)
+                item
+                for item in requires
+                if not E2EExpr.evaluate(item, features, build_only_mode)
+            ]
+        except ValueError as e:
+            raise ValueError("Error in REQUIRES list:\n%s" % str(e))
+
+    def getMatchedFromList(self, features, alist, build_only_mode=False):
+        try:
+            return [
+                item
+                for item in alist
+                if E2EExpr.evaluate(item, features, build_only_mode, False)
```

Comment:
This `False` implies that we're processing `UNSUPPORTED`, we need to modify the name of this function to indicate that now.